### DISCUSSION TITLE
Http Error 403 When Admin or External Guest Signs-out from PnP Search Page in .Aspx

### DISCUSSION THREAD

**Discussion by RamBalani** [microsoft-search/pnp-modern-search#2696] [Ideas]
![eSTARVelezYAHOO SignOff-Error Flashing When Signing Out](https://user-images.githubusercontent.com/37209670/212256058-fb9b1925-6941-4a95-8b63-a3d82d3a1a86.jpg)
We are getting an Http 403 Forbidden Access when accessing our Custom GuestSearch.Aspx on the Site Pages of one of our sites.
An Admin or External Guest gets a momentarily flashing message -see attachment screen show showing the exact Error message.
How do we resolve this so the Http Error 403 does not show up ?
Please help.
![Yahoo-External Access OTP Emailed Screen](https://user-images.githubusercontent.com/37209670/212255798-bc137755-5193-48ba-8e7f-fd50e700d018.jpg)

**Comment by wobba** [comment]
Added as an improvement to allow toggling showing of error message or not. Today it will always show.

**Reply by RamBalani** [reply]
What is the source of the error? 
Is it  Permission or access privileges related? This is a mystery to us.

Can you explain how to amend such that instead of mere toggling for error to show/or not depending on your so-called toggle-we actually fix the issue to avoid any errors? /

**Comment by RamBalani** [comment]
Thanks.
Can we implement the toggle on an existing  page containing our PnP Search Web parts?
How ? Where do we do that.
Appreciate any feedback you can share with this. BTW
My title is misleading it's an HTTP 403 not 404 though I reference it correctly in the message details-wills ee about amending this.


https://tinyurl.com/2p9ee4ba
[cid:d8b804ad-1abf-489f-b26e-087c74fc0761]
________________________________

**Reply by wobba** [reply]
The change requires a code change. Not possible in the templates. Once added we likely will have it default on like today and you have to edit the web part to toggle it off. Which means you have to edit page/web part. No plans to add this today, which means a community contributor in need of the feature can publish a PR with this.

**Comment by RamBalani** [comment]
@wobba -Noticed this CSS code snipped below on the official  PnP Search v4 Documentation just now--see below.

Can we not NOW include the CSS template code below  on our existing Page 2 PnP Search Result config 'Edit Layouts' to override the PnP template default and hide error messages? 
(* which as you said shows error messages by default*)
--------------------------------------------------------------------------------------------------------
"Hide error message using CSS[¶](https://microsoft-search.github.io/pnp-modern-search/extensibility/templating/#hide-error-message-using-css)

By default the web parts will output an error message if something goes wrong on for example API calls.

If you want to hide the error message you can add the following CSS in your custom template.

==========================================================
<content id="data-content">

    <style>
        .pnpSearchResultsErrorMessage { display:none; }
        ...
    </style>

    <div class="template">
        ...
    ..."
============================================
